experimental/DoubleAndHyperComplexes/src/Objects/Methods.jl

```diff
@@ -359,15 +359,13 @@ function _contract(
   @assert length(phi) == ngens(F0) "lengths are incompatible"
   result = zero(parent)
   n = ngens(F0)
-  for (i, ind) in enumerate(OrderedMultiIndexSet(p, n))
+  for (i, ind) in enumerate(combinations(n, p))
     is_zero(v[i]) && continue
     for j in 1:p
-      I = deleteat!(copy(indices(ind)), j)
-      new_ind = OrderedMultiIndex(I, n)
-      result = result + (-1)^j * v[i] * phi[ind[j]] * parent[linear_index(new_ind)]
+      I = deleteat!(copy(data(ind)), j)
```

Comment:
not a complaint, just a musing: why doesn't Julia have `deletat(data(ind),j)` (i.e. without the `!`)? *sigh*